src/rules/indentation/index.js

```diff
@@ -67,6 +67,7 @@ export default function (space, options = {}) {
       let nodeLevel = indentationLevel(node)
 
       if (options && options.hierarchicalSelectors) {
+        console.log("Option hierarchicalSelectors has been deprecated and will be removed in 7.0") // eslint-disable-line no-console
```

Comment:
FYI, we have a special deprecation warning type that the formatters treat differently. Example of usage: https://github.com/stylelint/stylelint/blob/master/src/postcssPlugin.js#L60-L66